editor/gui/editor_file_dialog.cpp

```diff
@@ -30,2661 +30,10 @@
-					}
-				} else {
-					valid = true;
-				}
-			}
-
-			// Add first extension of filter if no valid extension is found.
-			if (!valid) {
-				int idx = p_filter;
-				String flt = filters[idx].get_slicec(';', 0);
-				String ext = flt.get_slicec(',', 0).strip_edges().get_extension();
-				f += "." + ext;
-			}
-			emit_signal(SNAME("file_selected"), f);
-		} else if ((mode == FILE_MODE_OPEN_ANY || mode == FILE_MODE_OPEN_FILE) && dir_access->file_exists(f)) {
-			emit_signal(SNAME("file_selected"), f);
-		} else if (mode == FILE_MODE_OPEN_ANY || mode == FILE_MODE_OPEN_DIR) {
-			emit_signal(SNAME("dir_selected"), f);
-		}
-	}
-}
-
-void EditorFileDialog::popup_file_dialog() {
-	popup_centered_clamped(Size2(1050, 700) * EDSCALE, 0.8);
-	_focus_file_text();
-}
-
-void EditorFileDialog::_focus_file_text() {
-	int lp = file->get_text().rfind_char('.');
-	if (lp != -1) {
-		file->select(0, lp);
-		file->grab_focus();
-	}
-}
-
-VBoxContainer *EditorFileDialog::get_vbox() {
-	return vbox;
-}
-
-void EditorFileDialog::_update_theme_item_cache() {
-	ConfirmationDialog::_update_theme_item_cache();
-
-	theme_cache.parent_folder = get_editor_theme_icon(SNAME("ArrowUp"));
-	theme_cache.forward_folder = get_editor_theme_icon(SNAME("Forward"));
-	theme_cache.back_folder = get_editor_theme_icon(SNAME("Back"));
-	theme_cache.reload = get_editor_theme_icon(SNAME("Reload"));
-	theme_cache.toggle_hidden = get_editor_theme_icon(SNAME("GuiVisibilityVisible"));
-	theme_cache.toggle_filename_filter = get_editor_theme_icon(SNAME("FilenameFilter"));
-	theme_cache.favorite = get_editor_theme_icon(SNAME("Favorites"));
-	theme_cache.mode_thumbnails = get_editor_theme_icon(SNAME("FileThumbnail"));
-	theme_cache.mode_list = get_editor_theme_icon(SNAME("FileList"));
-	theme_cache.favorites_up = get_editor_theme_icon(SNAME("MoveUp"));
-	theme_cache.favorites_down = get_editor_theme_icon(SNAME("MoveDown"));
-	theme_cache.create_folder = get_editor_theme_icon(SNAME("FolderCreate"));
-	theme_cache.open_folder = get_editor_theme_icon(SNAME("FolderBrowse"));
-
-	theme_cache.filter_box = get_editor_theme_icon(SNAME("Search"));
-	theme_cache.file_sort_button = get_editor_theme_icon(SNAME("Sort"));
-
-	theme_cache.file = get_editor_theme_icon(SNAME("File"));
-	theme_cache.folder = get_editor_theme_icon(SNAME("Folder"));
-	theme_cache.folder_icon_color = get_theme_color(SNAME("folder_icon_color"), SNAME("FileDialog"));
-
-	theme_cache.action_copy = get_editor_theme_icon(SNAME("ActionCopy"));
-	theme_cache.action_delete = get_editor_theme_icon(SNAME("Remove"));
-	theme_cache.filesystem = get_editor_theme_icon(SNAME("Filesystem"));
-
-	theme_cache.folder_medium_thumbnail = get_editor_theme_icon(SNAME("FolderMediumThumb"));
-	theme_cache.file_medium_thumbnail = get_editor_theme_icon(SNAME("FileMediumThumb"));
-	theme_cache.folder_big_thumbnail = get_editor_theme_icon(SNAME("FolderBigThumb"));
-	theme_cache.file_big_thumbnail = get_editor_theme_icon(SNAME("FileBigThumb"));
```

Comment:
FileDialog uses big thumbnails by default. I don't think there is need for separate icons.